packages/storage/src/indexed-db/index.ts

```diff
@@ -63,11 +63,33 @@ import {
   DutchAuctionDescription,
 } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/component/auction/v1/auction_pb.js';
 import { ChainRegistryClient } from '@penumbra-labs/registry';
-import { PlainMessage } from '@bufbuild/protobuf';
+import { PartialMessage, PlainMessage } from '@bufbuild/protobuf';
 import { getAmountFromRecord } from '@penumbra-zone/getters/spendable-note-record';
 import { isZero } from '@penumbra-zone/types/amount';
 import { IDB_VERSION } from './config.js';
 
+const assertBytes = (v?: Uint8Array, expect?: number, name = 'value') => {
+  if (expect !== undefined && v?.length !== expect) {
+    throw new Error(`Expected ${name} of ${expect} bytes, but got ${v?.length} bytes`);
+  } else if (expect === undefined && !v?.length) {
+    throw new Error(`Expected ${name} to be non-empty, but got ${v?.length} bytes`);
+  }
+};
+
+// yes these are all 32 bytes
```

Comment:
Will they _always_ be 32 bytes in these cases forever?

Reply:
currently asset id's are set to 32 bytes and are internally represented as cryptographic [field elements](https://github.com/penumbra-zone/penumbra/blob/main/crates/core/asset/src/asset/id.rs#L31). Theoretically, this could change if we decide to switch cryptographic curves, though that would only be a long-term consideration.

Reply:
In general, I think we can simply rely upon the bufbuild types and not have to assert byte lengths to ensure the integrity of the data we are passed.

Reply:
that's a good point

Reply:
bufbuild classes do not validate length, and binary fields default init to a zero-length array. some validation is absolutely necessary